doc/tutorial/sam_file/sam_file_solution2.cpp

```diff
@@ -71,44 +40,47 @@ write_file_dummy_struct go{};
 
 int main()
 {
-    std::filesystem::path tmp_dir = std::filesystem::temp_directory_path(); // get the temp directory
+    std::filesystem::path current_path = std::filesystem::current_path();
 
     // read in reference information
-    seqan3::sequence_file_input reference_file{tmp_dir/"reference.fasta"};
-    std::vector<std::string> ref_ids{};
-    std::vector<seqan3::dna5_vector> ref_seqs{};
+    seqan3::sequence_file_input reference_file{current_path / "reference.fasta"};
+
+    std::vector<std::string> reference_ids{};
+    std::vector<seqan3::dna5_vector> reference_sequences{};
 
     for (auto && record : reference_file)
     {
-        ref_ids.push_back(std::move(record.id()));
-        ref_seqs.push_back(std::move(record.sequence()));
+        reference_ids.push_back(std::move(record.id()));
+        reference_sequences.push_back(std::move(record.sequence()));
     }
 
-    using field_type = seqan3::fields<seqan3::field::id,
-                                      seqan3::field::ref_id,
-                                      seqan3::field::mapq,
-                                      seqan3::field::alignment>;
-
-    seqan3::sam_file_input mapping_file{tmp_dir/"mapping.sam", ref_ids, ref_seqs, field_type{}};
+    // filter out alignments
+    seqan3::sam_file_input mapping_file{current_path / "mapping.sam", reference_ids, reference_sequences};
 
-    auto mapq_filter = std::views::filter([] (auto & rec) { return rec.mapping_quality() >= 30; });
+    auto mapq_filter = std::views::filter([] (auto & record) { return record.mapping_quality() >= 30; });
 
-    for (auto & [id, ref_id, mapq, alignment] : mapping_file | mapq_filter)
+    for (auto & record : mapping_file | mapq_filter)
     {
-        using seqan3::get;
-        size_t sum_ref{};
-        for (auto const & char_ref : get<0>(alignment))
-            if (char_ref == seqan3::gap{})
-                ++sum_ref;
+        using std::get;
+
+        // as loop
+        size_t sum_reference{};
+        for (auto const & char_reference : get<0>(record.alignment()))
+            if (char_reference == seqan3::gap{})
+                ++sum_reference;
+
+        // or via std::ranges::count
+        size_t sum_read = std::ranges::count(get<1>(record.alignment()), seqan3::gap{});
 
-        size_t sum_read{};
-        for (auto const & char_read : get<1>(alignment))
-            if (char_read == seqan3::gap{})
-                ++sum_read;
+        // The reference_id is ZERO based and an optional, where -1 will be represented as
+        // std::nullopt (= reference not known).
+        std::optional reference_id = record.reference_id();
 
-        seqan3::debug_stream << id << " mapped against " << ref_id << " with "
+        seqan3::debug_stream << record.id() << " mapped against "
+                             << (reference_id ? std::to_string(reference_id.value()) : "unknown reference")
```

Comment:
this is basically what the debug_stream would do (but we do it for verbosity)

Reply:
This was the discussion to show that reference_id is an optional and is zero based. We could also use the short one again :)

Reply:
Ah, I thought the comment was enough?
But "unknown reference" is probably nicer than what the debug stream would print

Reply:
yeah the debug stream would print `std::nullopt` I think

Reply:
`s << "<VALUELESS_VARIANT>";`